stan/math/rev/core/vari.hpp

```diff
@@ -280,29 +320,40 @@ class vari_value<T, std::enable_if_t<is_eigen_dense_base<T>::value>>
    * @param stacked If false will put this this vari on the nochain stack so
    * that its `chain()` method is not called.
    */
-  template <typename S, require_convertible_t<S&, value_type>* = nullptr>
+  template <typename S, require_convertible_t<S&, T_val>* = nullptr>
   vari_value(S&& x, bool stacked)
-      : val_mem_(ChainableStack::instance_->memalloc_.alloc_array<eigen_scalar>(
-            x.size())),
-        adj_mem_(ChainableStack::instance_->memalloc_.alloc_array<eigen_scalar>(
-            x.size())),
-        adj_(make_adj(x)),
-        val_(make_val(x)) {
+      : Base(),
+        adj_(plain_type_t<T_adj>::Zero(x.rows(), x.cols())),
+        val_(std::forward<S>(x)) {
     if (stacked) {
       ChainableStack::instance_->var_stack_.push_back(this);
     } else {
       ChainableStack::instance_->var_nochain_stack_.push_back(this);
     }
   }
 
+  /**
+   * Returns a view into a block of matrix.
+   * @param row starting row of the block
+   * @param col starting column of the block
+   * @param rows number of rows in the block
+   * @param cols number of columns in the block
+   * @return block
+   */
+  const vari_value<Eigen::Block<const T_val>, Eigen::Block<T_adj>> block(
+      Eigen::Index row, Eigen::Index col, Eigen::Index rows,
+      Eigen::Index cols) {
+    return {val_.block(row, col, rows, cols), adj_.block(row, col, rows, cols)};
+  }
```

Comment:
Let's have follow the eigen argument names
```suggestion
  /**
   * Returns a view into a block of matrix.
   * @param i starting row of the block
   * @param j starting column of the block
   * @param p number of rows in the block
   * @param q number of columns in the block
   * @return A vari_value containing a rectangular block of the `val_` and `adj_` matrices
   */
  const vari_value<Eigen::Block<const T_val>, Eigen::Block<T_adj>> block(
      Eigen::Index i, Eigen::Index j, Eigen::Index p,
      Eigen::Index q) const {
    return {val_.block(i, j, p, q), adj_.block(i, j, p, q)};
  }
```

Reply:
Noooo. I really don't like single letter variable names if I can think of better names for them. Especially if ther is many of them in one function.